code/game/objects/items/devices/illegalradio.dm

```diff
@@ -0,0 +1,177 @@
+/obj/item/device/illegalradio
+	icon = 'icons/obj/radio.dmi'
+	name = "black market uplink"
+	suffix = "\[3\]"
+	icon_state = "illegalradio"
+	item_state = "illegalradio"
+	desc = "A modified radio with a link to the black market. Use with caution."
+
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	throw_speed = 2
+	throw_range = 9
+	w_class = W_CLASS_SMALL
+	starting_materials = list(MAT_IRON = 75, MAT_GLASS = 25)
+	w_type = RECYK_ELECTRONIC
+	melt_temperature = MELTPOINT_PLASTIC
+	
+	var/money_stored		        // Money placed in the buffer
+	// List of items not to shove in their hands.
+	var/list/purchase_log = list()
+	var/show_description = null
```

Comment:
this is weird, fix it.
i assume this is just logic which you copy pasted, which would explain it not making any sense